--- build.gradle	2023-06-28 15:45:08.000000000 +0300
+++ build.gradle.in	2023-09-03 13:00:51.845554298 +0300
@@ -98,11 +98,12 @@
 
 android {
     
-    compileSdkVersion 26
+    namespace 'org.opencv'
+    compileSdk 34
 
     defaultConfig {
-        minSdkVersion 21
-        targetSdkVersion 26
+        minSdkVersion 24
+        targetSdkVersion 34
 
         versionCode openCVersionCode
         versionName openCVersionName
@@ -116,11 +117,14 @@
     }
 
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_17
+        targetCompatibility JavaVersion.VERSION_17
+    }
+
+    buildFeatures {
+        aidl true
     }
 
-    
 
     buildTypes {
         debug {
