diff --git a/build.gradle.orig b/build.gradle
index fab4e7c..e5ffe52 100644
--- a/build.gradle.orig
+++ b/build.gradle
@@ -99,11 +99,11 @@ println "OpenCV: " +openCVersionName + " " + project.buildscript.sourceFile
 
 android {
     namespace 'org.opencv'
-    compileSdkVersion 31
+    compileSdk 34
 
     defaultConfig {
-        minSdkVersion 21
-        targetSdkVersion 31
+        minSdk 24
+        targetSdk 34
 
         versionCode openCVersionCode
         versionName openCVersionName
@@ -117,8 +117,12 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_17
+        targetCompatibility JavaVersion.VERSION_17
+    }
+
+    lintOptions {
+        disable "MissingPermission"
     }
 
     buildTypes {
@@ -184,7 +188,7 @@ publishing {
     repositories {
         maven {
             name = 'myrepo'
-            url = "${project.buildDir}/repo"
+            url = "${rootProject.layout.buildDirectory}/repo"
         }
     }
 }
