--- sdk/build.gradle
+++ sdk/build.gradle
@@ -88,15 +88,11 @@
 //   }
 //
 
-
 apply plugin: 'com.android.library'
 apply plugin: 'maven-publish'
-try {
-    apply plugin: 'kotlin-android'
-    println "Configure OpenCV with Kotlin"
-} catch (Exception e) {
-    println "Configure OpenCV without Kotlin"
-}
+apply plugin: 'org.jetbrains.kotlin.android'
+// Kotlin is now built-in with AGP 9.0
+println "Configure OpenCV with Kotlin"
 
 def openCVersionName = "4.13.0"
 def openCVersionCode = ((4 * 100 + 13) * 100 + 0) * 10 + 0
@@ -105,11 +101,11 @@
 
 android {
     namespace 'org.opencv'
-    compileSdkVersion 34
+    compileSdkVersion 36
 
     defaultConfig {
         minSdkVersion 21
-        targetSdkVersion 34
+        targetSdkVersion 36
 
         versionCode openCVersionCode
         versionName openCVersionName
@@ -122,10 +118,18 @@
         }
     }
     
-    android {
-        buildFeatures {
-            buildConfig true
-        }
+    lint {
+        baseline = file("opencv-lint-baseline.xml")
+    }
+
+
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_21
+        targetCompatibility JavaVersion.VERSION_21
+    }
+
+    kotlinOptions {
+        jvmTarget = "21"
     }
 
     buildTypes {
@@ -139,7 +143,7 @@
                 doNotStrip '**/*.so'  // controlled by OpenCV CMake scripts
             }
             minifyEnabled false
-            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.txt'
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.txt'
         }
     }
 
@@ -193,7 +197,7 @@
     repositories {
         maven {
             name = 'myrepo'
-            url = "${project.buildDir}/repo"
+            url = "${rootProject.layout.buildDirectory}/repo"
         }
     }
 }
