diff --git a/build.gradle b/build.gradle.patched
index 5d176b4..586f683 100644
--- a/build.gradle
+++ b/build.gradle.patched
@@ -104,11 +104,11 @@ println "OpenCV: " +openCVersionName + " " + project.buildscript.sourceFile
 
 android {
     namespace 'org.opencv'
-    compileSdkVersion 34
+    compileSdkVersion 36
 
     defaultConfig {
         minSdkVersion 21
-        targetSdkVersion 34
+        targetSdkVersion 36
 
         versionCode openCVersionCode
         versionName openCVersionName
@@ -125,6 +125,10 @@ android {
         buildFeatures {
             buildConfig true
         }
+
+        lint {
+            baseline = file("opencv-lint-baseline.xml")
+        }
     }
     compileOptions {
         sourceCompatibility JavaVersion.VERSION_17
@@ -196,7 +200,7 @@ publishing {
     repositories {
         maven {
             name = 'myrepo'
-            url = "${project.buildDir}/repo"
+            url = "${rootProject.layout.buildDirectory}/repo"
         }
     }
 }
