--- build.orig.gradle.bak	2026-01-28 23:00:37.989164883 +0300
+++ build.gradle	2026-01-28 22:46:08.410973308 +0300
@@ -104,11 +104,11 @@
 
 android {
     namespace 'org.opencv'
-    compileSdkVersion 34
+    compileSdkVersion 36
 
     defaultConfig {
-        minSdkVersion 21
-        targetSdkVersion 34
+        minSdkVersion 24
+        targetSdkVersion 36
 
         versionCode openCVersionCode
         versionName openCVersionName
@@ -126,9 +126,18 @@
             buildConfig true
         }
     }
+
+     lint {
+         baseline = file("opencv-lint-baseline.xml")
+    }
+ 
+     kotlinOptions {
+         jvmTarget = "21"
+     }
+    
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_17
-        targetCompatibility JavaVersion.VERSION_17
+        sourceCompatibility JavaVersion.VERSION_21
+        targetCompatibility JavaVersion.VERSION_21
     }
 
     buildTypes {
@@ -142,7 +151,7 @@
                 doNotStrip '**/*.so'  // controlled by OpenCV CMake scripts
             }
             minifyEnabled false
-            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.txt'
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.txt'
         }
     }
 
@@ -196,7 +205,7 @@
     repositories {
         maven {
             name = 'myrepo'
-            url = "${project.buildDir}/repo"
+            url = "${rootProject.layout.buildDirectory}/repo"
         }
     }
 }
